src/v2/view-builder/internals/BaseFooter.js

```diff
@@ -24,7 +24,25 @@ export default View.extend({
 
   initialize () {
     const links = _.resultCtx(this, 'links', this);
+    let cancelFn = this.options.appState.get('currentState').cancel;
+    let cancelObj = {
+      'actionPath': 'cancel',
+      'label': 'Sign out',
+      'name': 'cancel',
+      'type': 'link'
+    };
+    const cancelInLinks = links.find(obj => obj.name === 'cancel');
```

Comment:
I'd say as long as `cancelFn` is function, we could just add `cancelObject` to the `links` list.
For the case (cancel link has been added in sub class) shall be code fix in subclass instead of doing smart check in BaseFooter.

Reply:
I was thinking about that. But I prefer this since cancel is getting added in the `BaseFooter`, it can also do the job of making sure if removes it from the views that don't cancel.

Reply:
we'd only add `cancel` when the `cancelFn` exists. What use case we need to remove it even when `cancelFn` exists? at that case, can we just dont add the link?
If worry about sub class add cancel link which shall not be, it's bug in sub class and need to be fix there.